icu4c/source/common/dictbe.cpp

```diff
@@ -1053,6 +1061,12 @@ CjkBreakEngine::CjkBreakEngine(DictionaryMatcher *adoptDictionary, LanguageType
     // Korean dictionary only includes Hangul syllables
     fHangulWordSet.applyPattern(UnicodeString(u"[\\uac00-\\ud7a3]"), status);
     fHangulWordSet.compact();
+    fNumberWordSet.applyPattern(UnicodeString(u"[:Nd:]"), status);
```

Comment:
as I mention in the Java PR, since fNumberWordSet and fOpenPunctuationSet are use exactly in the same place, we should merge them into one UnicodeSet fNumberOrOpenPunctuationSet w/ u"[[:Nd:][:Pi:][:Ps:]]" pattern

Reply:
Resolved